src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
1. It would consume a significant amount of memory.
2. It partially defeats the purpose of having unified storage.
3. It fails miserably if a node's id changes due to a change of IP. The existing code has this problem too.

I honestly cant figure out what you're trying to say in your second paragraph.